flamedisx/nest/lxe_blocks/quanta_splitting.py

```diff
@@ -2,110 +2,476 @@
+                                    bonus_arg=energy)
+                nel_mean = yields[0]
+                nq_mean = yields[1]
+                ex_ratio = yields[2]
+                alpha = 1. / (1. + ex_ratio)
+
+                p_ni = tfp.distributions.Normal(loc=nq_mean*alpha,
+                                                scale=tf.sqrt(nq_mean*alpha) + 1e-10).cdf(_ions_produced + 0.5) \
+                    - tfp.distributions.Normal(loc=nq_mean*alpha,
+                                               scale=tf.sqrt(nq_mean*alpha) + 1e-10).cdf(_ions_produced - 0.5)
+
+                p_nq = tfp.distributions.Normal(loc=nq_mean*alpha*ex_ratio,
+                                                scale=tf.sqrt(nq_mean*alpha*ex_ratio) + 1e-10).cdf(
+                                                    nq - _ions_produced + 0.5) \
+                    - tfp.distributions.Normal(loc=nq_mean*alpha*ex_ratio,
+                                               scale=tf.sqrt(nq_mean*alpha*ex_ratio) + 1e-10).cdf(
+                                                    nq - _ions_produced - 0.5)
+
+            recomb_p = self.gimme('recomb_prob', data_tensor=data_tensor, ptensor=ptensor,
+                                  bonus_arg=(nel_mean, nq_mean, ex_ratio))
+            skew = self.gimme('skewness', data_tensor=data_tensor, ptensor=ptensor,
+                              bonus_arg=nq_mean)
+            var = self.gimme('variance', data_tensor=data_tensor, ptensor=ptensor,
+                             bonus_arg=(nel_mean, nq_mean, recomb_p, _ions_produced))
+            width_corr = self.gimme('width_correction', data_tensor=data_tensor, ptensor=ptensor,
+                                    bonus_arg=skew)
+            mu_corr = self.gimme('mu_correction', data_tensor=data_tensor, ptensor=ptensor,
+                                 bonus_arg=(skew, var, width_corr))
+
+            mean = (tf.ones_like(_ions_produced, dtype=fd.float_type()) - recomb_p) * _ions_produced - mu_corr
+            std_dev = tf.sqrt(var) / width_corr
+
+            if self.is_ER:
+                owens_t_terms = 5
+            else:
+                owens_t_terms = 2
+
+            p_nel = tfp.distributions.TruncatedSkewGaussianCC(loc=mean, scale=std_dev,
+                                                              skewness=skew,
+                                                              limit=_ions_produced,
+                                                              owens_t_terms=owens_t_terms).prob(electrons_produced)
+
+            p_mult = p_nq * p_ni * p_nel
+
+            # Contract over ions_produced
+            p_final = tf.reduce_sum(p_mult, 3)
+
+            r_final = p_final * rate_vs_energy
+
+            r_final = tf.where(tf.math.is_nan(r_final),
+                               tf.zeros_like(r_final, dtype=fd.float_type()),
+                               r_final)
+
+            return r_final
+
+        def compute_single_energy_approx(args):
```

Comment:
This function looks very similar to `compute_single_energy_full`. Could they be one method with branching paths controlled by an argument?

Reply:
Done.